### DISCUSSION TITLE
Dynamic Dataloaders

### DISCUSSION THREAD

**Discussion by florianraiber** [Netflix/dgs-framework#432] [Q&A] [answered]
Hi,
i want do register dynamic batch dataloaders. Is there some hidden extension point to do this?

Thanks,
Florian

**Comment by paulbakker** [comment]
What is a *dynamic* batch dataloader?

**Reply by florianraiber** [reply]
We use the `@DgsCodeRegistry` to create fetchers at runtime. As discussed here: [DGS doc](https://netflix.github.io/dgs/advanced/dynamic-schemas/).
But we want our fetchers to use dataloaders of type `MappedBatchLoaderWithContext`. For each fetcher we can also create a dataloader at runtime, but we do not find a way to register them.

In other words: It's a dataloader registered at runtime. :-)

**Comment by florianraiber** [comment, marked as answer]
I solved my issue implementing a graphql-java fetcher and using `graphql.schema.DataFetchingEnvironment#getDataLoaderRegistry().register(<name>,<loader>)`.

**Reply by paulbakker** [reply]
sorry, I missed your previous comment. What you're doing is the correct thing, that's basically what we're doing internally in the framework as well.

**Comment by ellebrecht** [comment]
I tried registering auto-generated `MappedBatchLoader`s dynamically at runtime according to the instructions above, in order to get rid of the static boilerplate code annotated with `@DgsDataLoader`. 
The DataLoaders are sucessfully registered in `com.netflix.graphql.dgs.DgsDataFetchingEnvironment.getDataLoaderRegistry()`. They are never run however, and queries using them always time out. 
I can only suspect it has something to do with scheduling. Anything I'm missing here? 
Also, would you consider offering a way to cleanly register DataLoaders dynamically at runtime through the DGS API?

**Reply by voronovmax** [reply]
Hi @ellebrecht, have you succeeded to fix the issue? I came across the same.
